src/extensions/nginx-app-protect/monitoring/processor/nap.go

```diff
@@ -573,33 +491,40 @@ func (f *NAPConfig) getViolations(logger *logrus.Entry) []*models.ViolationData
 				name = string(decodedName)
 				value = fmt.Sprintf("actual header value: %s. matched header value: %s", string(decodedActualValue), string(decodedMatchedValue))
 			}
-
-			if isB64Decoded {
-				violation.ContextData = &models.ContextData{
-					Name:  name,
-					Value: value,
-				}
-				break
-			}
-
-			decodedValue, err := base64.StdEncoding.DecodeString(value)
-			if err != nil {
-				logger.Errorf("could not decode the URL Value %s for %v", value, f.SupportID)
-				break
-			}
-
-			violation.ContextData = &models.ContextData{Name: name, Value: string(decodedValue)}
 		case violationContextRequest:
-			var name, value string
+			isB64Decoded = true
 			if v.DefinedLength != "" {
 				name = fmt.Sprintf("Defined length: %s", v.DefinedLength)
 				value = fmt.Sprintf("Detected length: %s", v.DetectedLength)
 			} else if v.TotalLen != "" {
 				name = fmt.Sprintf("Total length: %s", v.TotalLen)
 				value = fmt.Sprintf("Total length limit: %s", v.TotalLenLimit)
 			}
+		default:
+			contextualized = false
+		}
 
-			violation.ContextData = &models.ContextData{Name: name, Value: value}
+		if contextualized {
+			if isB64Decoded {
+				name, value = populateNameValue(violation.Name, name, value)
+				violation.ContextData = &models.ContextData{
+					Name:  name,
+					Value: value,
+				}
+			} else {
+				decodedName, err := base64.StdEncoding.DecodeString(name)
+				if err != nil {
+					logger.Errorf("could not decode the %s Name %s for %v", v.Context, name, f.SupportID)
+				} else {
+					decodedValue, err := base64.StdEncoding.DecodeString(value)
+					if err != nil {
+						logger.Errorf("could not decode the %s Value %s for %v", v.Context, value, f.SupportID)
+					} else {
+						name, value = populateNameValue(violation.Name, string(decodedName), string(decodedValue))
+						violation.ContextData = &models.ContextData{Name: name, Value: value}
+					}
+				}
```

Comment:
Is it possible to simplify this by flattening the error handling? (optional)

```suggestion
				decodedName, err := base64.StdEncoding.DecodeString(name)
				if err != nil {
					logger.Errorf("could not decode the %s Name %s for %v", v.Context, name, f.SupportID)
					return
				} 
				
				decodedValue, err := base64.StdEncoding.DecodeString(value)
				if err != nil {
					logger.Errorf("could not decode the %s Value %s for %v", v.Context, value, f.SupportID)
					return
				}
				
				name, value = populateNameValue(violation.Name, string(decodedName), string(decodedValue))
				violation.ContextData = &models.ContextData{Name: name, Value: value}
```

Reply:
If you want to only decode value if the `name` can be decoded and need to `populateNameValues`/`ContextData` unconditionally this might not be possible. Feel free to ignore

Reply:
Thanks for the comment Dean, yep that's what I want to achieve `only decode value if the name can be decoded and need to populateNameValues/ContextData unconditionally`. I will resolve the issue